Source/MyDoorGame/DoorActor.h

```diff
@@ -26,19 +27,23 @@ class MYDOORGAME_API ADoorActor : public AActor
 	// Called every frame
 	virtual void Tick(float DeltaTime) override;
 	
-	virtual void OnInteract(); 
-	class AMyDoorGameCharacter* Character; 
+	virtual void OnInteract(AMyDoorGameCharacter* User);
+
+	//UPROPERTY(Transient)
```

Comment:
In this implementation, you are just using the Character temporary so there's no need to cache it as a member function.  But I still want to show you the proper way of keep a TRANSIENT (not seralized) reference to a uobject. 

If in some case using UPROPERTY(Transient) is not possible you can also do TWeakObjectPtr<>